mono/metadata/marshal.c

```diff
@@ -1358,11 +1383,6 @@ mono_marshal_get_ptr_to_stringbuilder_conv (MonoMethodPInvoke *piinfo, MonoMarsh
 
 	switch (encoding) {
 	case MONO_NATIVE_LPWSTR:
-		/* 
-		 * mono_string_builder_to_utf16 does not allocate a 
-		 * new buffer, so no need to free it.
-		 */
-		*need_free = FALSE;
```

Comment:
There are cases where we can skip the allocation in `mono_string_builder_to_utf16`. However, that is currently not done and hence the comment was wrong. I kept the `need_free` argument in case someone decides to optimize this code path later.